include/sacn/universe_discovery.h

```diff
@@ -0,0 +1,253 @@
+ *
+ * This callback is rate-limited: it will only be called when the first universe discovery packet is received that takes
+ * the module beyond a memory limit.  After that, it will not be called until the number of sources or universes has
+ * dropped below the limit and hits it again.
+ *
+ * @param[in] handle Handle to the listening universe discovery instance.
+ * @param[in] context Context pointer that was given at the creation of the universe discovery instance.
+ */
+typedef void (*SacnUniverseDiscoveryLimitExceededCallback)(sacn_universe_discovery_t handle, void* context);
+
+/** A set of callback functions that the library uses to notify the application about universe discovery events. */
+typedef struct SacnUniverseDiscoveryCallbacks
+{
+  SacnUniverseDiscoveryUpdateSourceCallback source_update;   /**< Required */
+  SacnUniverseDiscoverySourceExpiredCallback source_expired; /**< Required */
+  SacnUniverseDiscoveryLimitExceededCallback limit_exceeded; /**< Optional */
+  void* context; /**< (optional) Pointer to opaque data passed back with each callback. */
+} SacnUniverseDiscoveryCallbacks;
+
+/** A set of configuration information for a sACN Universe Discovery listener. */
+typedef struct SacnUniverseDiscoveryConfig
+{
+  /** The callbacks this listener will use to notify the application of events. */
+  SacnUniverseDiscoveryCallbacks callbacks;
+
+  /********* Optional values **********/
+
+  /** The maximum number of sources this listener will record.  It is recommended that applications using dynamic
+     memory use #SACN_UNIVERSE_DISCOVERY_INFINITE for this value. This parameter is ignored when configured to use
+     static memory -- #SACN_UNIVERSE_DISCOVERY_MAX_SOURCES is used instead.*/
+  size_t source_count_max;
+
+  /** The maximum number of universes this listener will record for a source.  It is recommended that applications using
+     dynamic memory use #SACN_UNIVERSE_DISCOVERY_INFINITE for this value. This parameter is ignored when configured to
+     use static memory -- #SACN_UNIVERSE_DISCOVERY_MAX_SOURCES is used instead.*/
+  size_t universes_per_source_max;
+} SacnUniverseDiscoveryConfig;
+
+/** A default-value initializer for an SacnUniverseDiscoveryConfig struct. */
+#define SACN_UNIVERSE_DISCOVERY_CONFIG_DEFAULT_INIT                                               \
+  {                                                                                               \
+    {NULL, NULL, NULL, NULL}, SACN_UNIVERSE_DISCOVERY_INFINITE, SACN_UNIVERSE_DISCOVERY_INFINITE, \
+  }
+
+void sacn_universe_discovery_config_init(SacnUniverseDiscoveryConfig* config);
+
+etcpal_error_t sacn_universe_discovery_create(const SacnUniverseDiscoveryConfig* config,
+                                              sacn_universe_discovery_t* handle, SacnMcastInterface* netints,
+                                              size_t num_netints);
+etcpal_error_t sacn_universe_discovery_destroy(sacn_universe_discovery_t handle);
```

Comment:
I've seen other destroy functions return void. Can we do the same here?

Reply:
Must have been a copy paste error.  I agree.